heron/common/src/cpp/network/event_loop_impl.cpp

```diff
@@ -18,6 +18,7 @@
  */
 
 #include "network/event_loop_impl.h"
+#include <signal.h>
```

Comment:
What is this change needed for?

Reply:
[signals.h](https://man7.org/linux/man-pages/man0/signal.h.0p.html) has a bunch of macros and methods for handling interrupts signals from the OS. `SIGTERM` is one such macro.

https://github.com/apache/incubator-heron/blob/b248f767c636a0b114a7b0ba3bad433c97189303/heron/common/src/cpp/network/event_loop_impl.cpp#L86

Not certain why it would now need to now be explicitly included unless the updates to the Protobuf mean these definitions are no longer included in one of the other headers (compilation failures will result).

Reply:
Yup, confirmed from the build log:

```bash
  /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer '-std=c++0x' -MD -MF bazel-out/k8-fastbuild/bin/heron/common/src/cpp/network/_objs/network-cxx/event_loop_impl.pic.d '-frandom-seed=bazel-out/k8-fastbuild/bin/heron/common/src/cpp/network/_objs/network-cxx/event_loop_impl.pic.o' -fPIC -iquote . -iquote bazel-out/k8-fastbuild/bin -iquote external/com_github_google_glog -iquote bazel-out/k8-fastbuild/bin/external/com_github_google_glog -iquote external/com_github_gflags_gflags -iquote bazel-out/k8-fastbuild/bin/external/com_github_gflags_gflags -iquote external/com_github_gperftools_gperftools -iquote bazel-out/k8-fastbuild/bin/external/com_github_gperftools_gperftools -iquote external/com_github_corvusoft_kashmir_cpp -iquote bazel-out/k8-fastbuild/bin/external/com_github_corvusoft_kashmir_cpp -iquote external/com_google_protobuf -iquote bazel-out/k8-fastbuild/bin/external/com_google_protobuf -iquote external/zlib -iquote bazel-out/k8-fastbuild/bin/external/zlib -iquote external/org_nongnu_libunwind -iquote bazel-out/k8-fastbuild/bin/external/org_nongnu_libunwind -iquote external/org_libevent_libevent -iquote bazel-out/k8-fastbuild/bin/external/org_libevent_libevent -Ibazel-out/k8-fastbuild/bin/external/com_github_google_glog/_virtual_includes/glog -Ibazel-out/k8-fastbuild/bin/external/com_github_gflags_gflags/_virtual_includes/gflags -isystem external/com_github_gperftools_gperftools/include -isystem bazel-out/k8-fastbuild/bin/external/com_github_gperftools_gperftools/include -isystem external/com_google_protobuf/src -isystem bazel-out/k8-fastbuild/bin/external/com_google_protobuf/src -isystem external/zlib/zlib/include -isystem bazel-out/k8-fastbuild/bin/external/zlib/zlib/include -isystem external/org_nongnu_libunwind/include -isystem bazel-out/k8-fastbuild/bin/external/org_nongnu_libunwind/include -isystem external/org_libevent_libevent/include -isystem bazel-out/k8-fastbuild/bin/external/org_libevent_libevent/include -O3 -Iheron/common/src/cpp -Ibazel-out/k8-fastbuild/bin/heron/common/src/cpp -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c heron/common/src/cpp/network/event_loop_impl.cpp -o bazel-out/k8-fastbuild/bin/heron/common/src/cpp/network/_objs/network-cxx/event_loop_impl.pic.o)
Execution platform: @local_config_platform//:host
In file included from external/org_libevent_libevent/include/event.h:69,
                 from heron/common/src/cpp/network/regevent.h:27,
                 from heron/common/src/cpp/network/event_loop_impl.cpp:26:
heron/common/src/cpp/network/event_loop_impl.cpp: In member function 'virtual void EventLoopImpl::loop()':
heron/common/src/cpp/network/event_loop_impl.cpp:85:50: error: 'SIGTERM' was not declared in this scope
   85 |   struct event *term = evsignal_new(mDispatcher, SIGTERM, handleTerm, mDispatcher);
      |                                                  ^~~~~~~
```

Reply:
```
Execution platform: @local_config_platform//:host
In file included from external/org_libevent_libevent/include/event.h:69:0,
                 from heron/common/src/cpp/network/regevent.h:27,
                 from heron/common/src/cpp/network/event_loop_impl.cpp:26:
heron/common/src/cpp/network/event_loop_impl.cpp: In member function 'virtual void EventLoopImpl::loop()':
heron/common/src/cpp/network/event_loop_impl.cpp:85:50: error: 'SIGTERM' was not declared in this scope
   struct event *term = evsignal_new(mDispatcher, SIGTERM, handleTerm, mDispatcher);
                                                  ^
```

Reply:
Awesome, thanks for the extra context.